src/components/new-safe/load/steps/SetAddressStep/index.tsx

```diff
@@ -0,0 +1,161 @@
+import type { StepRenderProps } from '@/components/new-safe/CardStepper/useCardStepper'
+import type { LoadSafeFormData } from '@/components/new-safe/load'
+import { FormProvider, useForm } from 'react-hook-form'
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Divider,
+  Grid,
+  InputAdornment,
+  SvgIcon,
+  Tooltip,
+  Typography,
+} from '@mui/material'
+import layoutCss from '@/components/new-safe/create/styles.module.css'
+import NameInput from '@/components/common/NameInput'
+import InfoIcon from '@/public/images/notifications/info.svg'
+import css from '@/components/new-safe/create/steps/Step1/styles.module.css'
+import NetworkSelector from '@/components/common/NetworkSelector'
+import { useMnemonicSafeName } from '@/hooks/useMnemonicName'
+import { useAddressResolver } from '@/hooks/useAddressResolver'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import AddressInput from '@/components/common/AddressInput'
+import React from 'react'
+import { getSafeInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import useChainId from '@/hooks/useChainId'
+import { useAppSelector } from '@/store'
+import { selectAddedSafes } from '@/store/addedSafesSlice'
+import { LOAD_SAFE_EVENTS, trackEvent } from '@/services/analytics'
+import ExternalLink from '@/components/common/ExternalLink'
+
+enum Field {
+  name = 'name',
+  address = 'address',
+}
+
+type FormData = {
+  [Field.name]: string
+  [Field.address]: string
+}
+
+const SetAddressStep = ({ data, onSubmit, onBack }: StepRenderProps<LoadSafeFormData>) => {
+  const currentChainId = useChainId()
+  const addedSafes = useAppSelector((state) => selectAddedSafes(state, currentChainId))
+  const formMethods = useForm<FormData>({
+    mode: 'all',
+    defaultValues: {
+      [Field.name]: data.name,
+      [Field.address]: data.address,
+    },
+  })
+
+  const {
+    handleSubmit,
+    formState: { errors, isValid },
+    watch,
+    getValues,
+  } = formMethods
+
+  const safeAddress = watch(Field.address)
+
+  const randomName = useMnemonicSafeName()
+  const { ens, name, resolving } = useAddressResolver(safeAddress)
+
+  // Address book, ENS, mnemonic
+  const fallbackName = name || ens || randomName
+
+  const validateSafeAddress = async (address: string) => {
+    if (addedSafes && Object.keys(addedSafes).includes(address)) {
+      return 'Safe is already added'
+    }
+
+    try {
+      await getSafeInfo(currentChainId, address)
```

Comment:
async validation is not handled that well here.
If `getSafeInfo` takes a bit longer (which it sometimes does) the Next button is disabled without explanation why. 
We should add a spinner to the `AddressInput` to have an indicator that the validation is in progress. 

This would also solve #1203

Reply:
I fixed it in a separate fix PR. So nevermind this comment :)